apps/sms/views/shared/js/activity_picker.js

```diff
@@ -3,87 +3,119 @@
+     * @returns {Promise} DOMRequest promise is always successfully resolved
+     */
+    url(url) {
+      return Promise.resolve(new MozActivity({
+        name: 'view',
+        data: {
+          type: 'url',
+          url: url
+        }
+      }));
+    },
+
+    /**
+     * When user wants to create a new contact entry.
+     * @param  {Object} Contact object.
+     * @returns {Promise} DOMRequest promise is always successfully resolved
+     */
+    createNewContact(contactProps) {
+      return Promise.resolve(new MozActivity({
+        name: 'new',
+        data: {
+          type: 'webcontacts/contact',
+          params: contactProps
+        }
+      }));
+    },
 
-var ActivityPicker = {
-  dial: function ap_call(number, onsuccess, onerror) {
-    handleActivity(new MozActivity({
-      name: 'dial',
-      data: {
-        type: 'webtelephony/number',
-        number: number
-      }
-    }), onsuccess, onerror);
-  },
-  email: function ap_email(email, onsuccess, onerror) {
-    handleActivity(new MozActivity({
-      name: 'new',
-      data: {
-        type: 'mail',
-        URI: 'mailto:' + email
-      }
-    }), onsuccess, onerror);
-  },
-  url: function ap_browse(url, onsuccess, onerror) {
-    handleActivity(new MozActivity({
-      name: 'view',
-      data: {
-        type: 'url',
-        url: url
-      }
-    }), onsuccess, onerror);
-  },
-  createNewContact: function ap_createNewContact(contactProps) {
-    handleActivity(new MozActivity({
-      name: 'new',
-      data: {
-        type: 'webcontacts/contact',
-        params: contactProps
-      }
-    }));
-  },
-  addToExistingContact: function ap_addToExistingContact(contactProps) {
-    handleActivity(new MozActivity({
-      name: 'update',
-      data: {
-        type: 'webcontacts/contact',
-        params: contactProps
-      }
-    }));
-  },
-  viewContact: function ap_viewContact(contactProps) {
-    handleActivity(new MozActivity({
-      name: 'open',
-      data: {
-        type: 'webcontacts/contact',
-        params: contactProps
-      }
-    }));
-  },
-  openSettings: function ap_openSettings(onsuccess, onerror) {
-    handleActivity(new MozActivity({
-      name: 'configure',
-      data: {
-        target: 'device',
-        section: 'messaging'
-      }
-    }), onsuccess, onerror);
-  }
-};
+    /**
+     * When user wants to update a contact.
+     * @param  {Object} Contact object.
+     * @returns {Promise} DOMRequest promise is always successfully resolved
+     */
+    addToExistingContact(contactProps) {
+      return Promise.resolve(new MozActivity({
+        name: 'update',
+        data: {
+          type: 'webcontacts/contact',
+          params: contactProps
+        }
+      }));
+    },
+
+    /**
+     * When user wants to open a contact.
+     * @param  {Object} Contact object.
+     * @returns {Promise} DOMRequest promise is always successfully resolved
+     */
+    viewContact(contactProps) {
```

Comment:
Like I said previously, the contactProps here actually contains {id: opt.contactId}